src/clr-angular/popover/signpost/providers/signpost-id.service.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ *  Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ *
+ */
+
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { Subject } from 'rxjs';
+
+@Injectable()
+export class SignpostIdService {
+  private _id: Subject<string> = new Subject<string>();
+
+  setId(id: string) {
+    this._id.next(id);
```

Comment:
Why is this `setId()` and not `set id():`?

Reply:
Getters and setters in typescript need to have the same type. I'm using a getter for the Observable to have cleaner syntax when subscribing and that means I can't have  a setter with a boolean arg.